pkg/common/common.go

```diff
@@ -216,3 +217,20 @@ func StripDefaultPorts(fromUrl string) (string, error) {
 	}
 	return r.URL.String(), nil
 }
+
+// GetImagePullPolicy get imagePullPolicy for a container, based on its image.
+// If image contains a sha256 or sha512 digest, use IfNotPresent; otherwise, Always.
+// Reference: https://github.com/distribution/distribution/blob/v2.7.1/reference/reference.go
+func GetImagePullPolicy(image string) corev1.PullPolicy {
+	matched, err := regexp.Match("@sha256:[a-f0-9]{64}", []byte(image))
```

Comment:
lets log the error here or you can ignore the returned error `matched, _ := regexp.Match...` or even better lets use `regex.Compile` first then use `regex.Match`

Reply:
going to take a look in `regex.Compile`

Reply:
done

had to change more things then I though :cry: 

But tested and looks it is working. Example log from oadp controller manager when changing regex wrongly 
```
2024-06-10T18:14:05Z	ERROR	imagePullPolicy regex failed	{"controller": "dataprotectionapplication", "controllerGroup": "oadp.openshift.io", "controllerKind": "DataProtectionApplication", "DataProtectionApplication": {"name":"velero-sample","namespace":"test-oadp-operator"}, "namespace": "test-oadp-operator", "name": "velero-sample", "reconcileID": "c306a4fd-3d41-48a2-ad38-fadef0e6da2d", "error": "error parsing regexp: missing closing ]: `[a-f0-9{64}`"}
github.com/openshift/oadp-operator/controllers.(*DPAReconciler).customizeNodeAgentDaemonset
	/go/src/github.com/openshift/oadp-operator/controllers/nodeagent.go:335
github.com/openshift/oadp-operator/controllers.(*DPAReconciler).buildNodeAgentDaemonset
	/go/src/github.com/openshift/oadp-operator/controllers/nodeagent.go:228
github.com/openshift/oadp-operator/controllers.(*DPAReconciler).ReconcileNodeAgentDaemonset.func1
	/go/src/github.com/openshift/oadp-operator/controllers/nodeagent.go:145
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/controller/controllerutil/controllerutil.go:426
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrPatch
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/controller/controllerutil/controllerutil.go:361
github.com/openshift/oadp-operator/controllers.(*DPAReconciler).ReconcileNodeAgentDaemonset
	/go/src/github.com/openshift/oadp-operator/controllers/nodeagent.go:123
github.com/openshift/oadp-operator/controllers.ReconcileBatch
	/go/src/github.com/openshift/oadp-operator/controllers/dpa_controller.go:218
github.com/openshift/oadp-operator/controllers.(*DPAReconciler).Reconcile
	/go/src/github.com/openshift/oadp-operator/controllers/dpa_controller.go:92
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:227
```